apm-sniffer/apm-agent/src/main/java/org/apache/skywalking/apm/agent/SkyWalkingAgent.java

```diff
@@ -215,4 +218,23 @@ public void onError(final String typeName,
         public void onComplete(String typeName, ClassLoader classLoader, JavaModule module, boolean loaded) {
         }
     }
+
+    private static class RedefinitionListener implements AgentBuilder.RedefinitionStrategy.Listener {
+
+        @Override
+        public void onBatch(int index, List<Class<?>> batch, List<Class<?>> types) {
+            /* do nothing */
+        }
+
+        @Override
+        public Iterable<? extends List<Class<?>>> onError(int index, List<Class<?>> batch, Throwable throwable, List<Class<?>> types) {
+            LOGGER.error(throwable, "index={}, batch={}, types={}", index, batch, types);
+            return Collections.emptyList();
```

Comment:
I am not sure about this `onError` API, do we need to return `batch` or empty list?

Reply:
according to the api doc:
> Returns:
A set of classes which should be attempted to be redefined. Typically, this should be a subset of the classes contained in batch but not all classes.

We don't need to attempt to redefine again.
The default listener is net.bytebuddy.agent.builder.AgentBuilder.RedefinitionStrategy.Listener.NoOp.I think adding the error log is enough.